pkg/tfbridge/provider.go

```diff
@@ -664,7 +664,7 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 	// Now fetch the default values so that (a) we can return them to the caller and (b) so that validation
 	// includes the default values.  Otherwise, the provider wouldn't be presented with its own defaults.
 	tfname := res.TFName
-	inputs, assets, err := MakeTerraformInputs(ctx,
+	inputs, _, err := MakeTerraformInputsNoTFDefaults(ctx,
```

Comment:
call this "inputsWithoutDefaults"?

Reply:
I notice only Check method is changing in this PR which is great (keep changes small), but I am curious what your sense is if this problem also applies to CheckConfig (validating provider-level properties) and Invoke (validating function arguments). If we think it might apply as well worth backlogging?

Reply:
It seems quite likely that the same issues happen there but I was not able to find any concrete issues in the providers which exhibit this for either `CheckConfig` or `Invoke`.

I'll make issues for them and we can revisit.